src/hostcalls.rs

```diff
@@ -1506,25 +1506,34 @@ fn get_hostfunc(
                     println!(
                         "[vm->host] proxy_grpc_call(service={service}, service_name={service_name}, method_name={method_name}, initial_metadata={initial_metadata:?}, request={request:?}, timeout={timeout_milliseconds}");
 
-                    let token_id = match EXPECT.lock().unwrap().staged.get_expect_grpc_call(
-                        service,
-                        service_name,
-                        method_name,
-                        initial_metadata,
-                        request,
-                        timeout_milliseconds,
-                    ) {
-                        Some(expect_token) => expect_token,
-                        None => 0,
-                    };
-
-                    unsafe {
-                        let return_token_add = mem.data_mut(&mut caller).get_unchecked_mut(
-                            token_ptr as u32 as usize..token_ptr as u32 as usize + 4,
-                        );
-                        return_token_add.copy_from_slice(&token_id.to_le_bytes());
+                    let call_result = EXPECT
+                        .lock()
+                        .unwrap()
+                        .staged
+                        .get_expect_grpc_call(
+                            service,
+                            service_name,
+                            method_name,
+                            initial_metadata,
+                            request,
+                            timeout_milliseconds,
+                        )
+                        .unwrap_or(Ok(0));
```

Comment:
You seem to change this for a `Err` now, to get the previous behavior... that was already confusing, but that's now a new pattern, I don't think this makes sense...

Reply:
So why the `Result` here? if the `Err` path gets dropped on the floor?

Reply:
It get rewrapped in an `Option::Some` always... Now I get it

Reply:
... other than when it isn't expected...